Util/model_parser/model_parser.H

```diff
@@ -102,73 +109,35 @@ AMREX_INLINE AMREX_GPU_HOST_DEVICE
-           (model::profile(model_index).r(id+1) - model::profile(model_index).r(id));
-       interp = slope * (r - model::profile(model_index).r(id)) + model::profile(model_index).state(id, var_index);
-
-       // safety check to make sure interp lies within the bounding points
-       Real minvar = amrex::min(model::profile(model_index).state(id+1, var_index),
-                                model::profile(model_index).state(id, var_index));
-       Real maxvar = amrex::max(model::profile(model_index).state(id+1, var_index),
-                                model::profile(model_index).state(id, var_index));
-       interp = amrex::max(interp, minvar);
-       interp = amrex::min(interp, maxvar);
-
-    } else if (id == model::npts-1) {
-
-       slope = (model::profile(model_index).state(id, var_index) -
-                model::profile(model_index).state(id-1, var_index)) /
-           (model::profile(model_index).r(id) - model::profile(model_index).r(id-1));
-       interp = slope * (r - model::profile(model_index).r(id)) + model::profile(model_index).state(id, var_index);
-
-
-       // safety check to make sure interp lies within the bounding points
-       Real minvar = amrex::min(model::profile(model_index).state(id-1, var_index),
-                                model::profile(model_index).state(id, var_index));
-       Real maxvar = amrex::max(model::profile(model_index).state(id-1, var_index),
-                                model::profile(model_index).state(id, var_index));
-       interp = amrex::max(interp, minvar);
-       interp = amrex::min(interp, maxvar);
-
-    } else {
+    slope = (model::profile(model_index).state(id+1, var_index) -
+             model::profile(model_index).state(id, var_index)) /
+        (model::profile(model_index).r(id+1) - model::profile(model_index).r(id));
+    interp = slope * (r - model::profile(model_index).r(id)) +
+        model::profile(model_index).state(id, var_index);
 
-        if (r >= model::profile(model_index).r(id)) {
-
-            slope = (model::profile(model_index).state(id+1, var_index) -
-                     model::profile(model_index).state(id, var_index)) /
-                (model::profile(model_index).r(id+1) - model::profile(model_index).r(id));
-            interp = slope * (r - model::profile(model_index).r(id)) + model::profile(model_index).state(id, var_index);
-
-        } else {
-
-            slope = (model::profile(model_index).state(id, var_index) -
-                     model::profile(model_index).state(id-1, var_index)) /
-                (model::profile(model_index).r(id) - model::profile(model_index).r(id-1));
-            interp = slope * (r - model::profile(model_index).r(id)) + model::profile(model_index).state(id, var_index);
-
-        }
-
-    }
+    // safety check to make sure interp lies within the bounding points
+    Real minvar = std::min(model::profile(model_index).state(id+1, var_index),
```

Comment:
Do we want this for r < profile.r(0)?

Reply:
indeed, probably not.  That's a good point.

Reply:
updated